mysql-test/main/information_schema_temp_table.test

```diff
@@ -278,3 +278,19 @@ drop table seq2;
 show  full tables;
 drop table seq3;
 show  full tables;
+
+--echo #
+--echo # MDEV-32973: SHOW TABLES LIKE shows temporary tables with
+--echo #             non-matching names
+--echo #
+
+CREATE DATABASE db;
+USE db;
+CREATE TABLE realtable (a INT);
+CREATE TEMPORARY TABLE temptable (a INT);
+SHOW TABLES LIKE 'faketable';
```

Comment:
add:
```sql
SHOW TABLES LIKE '%pt%';
```
and also you can try with `--lower-case-table-names=1` (in a separate test file)
```sql
create table t1 (a int);
select table_name from information_schema.tables where table_schema='test' and table_name='T1'
```